lib/identity_cache/cache_key_generation.rb

```diff
@@ -7,14 +7,17 @@ def self.schema_to_string(columns)
       columns.sort_by(&:name).map{|c| "#{c.name}:#{c.type}"}.join(',')
     end
 
-    def self.denormalized_schema_hash(klass)
+    def self.denormalized_schema_hash(klass, seen_assocs = Set.new)
       schema_string = schema_to_string(klass.columns)
       klass.send(:all_cached_associations).sort.each do |name, options|
         klass.send(:check_association_scope, name)
         ParentModelExpiration.check_association(options) if options[:embed]
         case options[:embed]
         when true
-          schema_string << ",#{name}:(#{denormalized_schema_hash(options[:association_reflection].klass)})"
+          unless seen_assocs.add?(name)
```

Comment:
You would need to use the association klass as well as the name.  Otherwise, this might have false positives where it finds two associations on different classes with the same name and assume the association has already been added when it hasn't been.